src/hooks/useApi.ts

```diff
@@ -97,16 +99,27 @@ export const useServerApi = () => {
     authRequired = false,
     impersonationData,
     version,
-  }: Omit<RequestOptions, 'additionalHeaders'> & {
+    additionalHeaders,
+  }: RequestOptions & {
     impersonationData?: ImpersonationData;
     version?: string;
   }): Promise<T> => {
     const headers: Record<string, string> = {
-      'x-locale': lang,
+      'x-locale': locale,
       'tenant-key': tenantConfig?.id || '',
       'X-SESSION-ID': await getSessionId(),
+      ...(additionalHeaders ? additionalHeaders : {}),
     };
 
+    // Only add version header if version is explicitly provided
+    if (version !== undefined) {
+      headers['x-accept-versions'] = version || '1.0.3';
+    }
+    // Set 'Content-Type' to 'application/json' only for  requests that send a body
+    if (['POST', 'PUT', 'DELETE'].includes(method)) {
+      headers['Content-Type'] = 'application/json';
+    }
```

Comment:
1. I suggest keeping this logic in `apiClient` as it directly tied to the request body format.
2. Is this needed for `DELETE`?

Reply:
As of now, we are not sending a body for the DELETE method, so I have updated the logic to avoid setting the Content-Type header for it.

Reply:
https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/2455/commits/1604a9a20739f754229d17bc02c91a5cfd0e2cbe